apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/valueset/ValueSetReader.java

```diff
@@ -20,87 +20,91 @@
 @Service
 public class ValueSetReader {
 
-    @Autowired
-    private ValueSetRepository valueSetRepository;
-
-    @Autowired
-    private CodeValueGeneralRepository codeValueGeneralRepository;
-
-    public Page<ValueSet> findAllValueSets(Pageable pageable) {
-        Page<Codeset> rawResults = valueSetRepository.findAll(pageable);
-        List<ValueSet> formatResults = toReadValueSet(rawResults);
-        return new PageImpl<>(formatResults, pageable, rawResults.getTotalElements());
-
-    }
-
-    public Page<ValueSet> searchValueSearch(ValueSetSearchRequest search, Pageable pageable) {
-        Page<Codeset> rawResults = valueSetRepository.findByCodeSetNmOrValueSetNmorValueSetCodeorValueSetType(search.getCodeSetName(),
-                search.getValueSetNm(), search.getValueSetCode(), search.getValueSetTypeCd(), search.getValueSetDescription(), pageable);
-        List<ValueSet> formatResults = toReadValueSet(rawResults);
-        return new PageImpl<>(formatResults, pageable, rawResults.getTotalElements());
+  private final ValueSetRepository valueSetRepository;
+
+  private final CodeValueGeneralRepository codeValueGeneralRepository;
+
+  ValueSetReader(final ValueSetRepository valueSetRepository,
+      final CodeValueGeneralRepository codeValueGeneralRepository) {
+    this.valueSetRepository = valueSetRepository;
+    this.codeValueGeneralRepository = codeValueGeneralRepository;
+  }
+
+  public Page<ValueSet> findAllValueSets(Pageable pageable) {
+    Page<Codeset> rawResults = valueSetRepository.findAll(pageable);
+    List<ValueSet> formatResults = toReadValueSet(rawResults);
+    return new PageImpl<>(formatResults, pageable, rawResults.getTotalElements());
+
+  }
+
+  public Page<ValueSet> searchValueSet(ValueSetSearchRequest search, Pageable pageable) {
+    Page<Codeset> rawResults = valueSetRepository.findByValueSetNmOrValueSetCodeOrValueSetDescription(
+        search.getValueSetNm(), search.getValueSetCode(), search.getValueSetDescription(), pageable);
+    List<ValueSet> formatResults = toReadValueSet(rawResults);
+    return new PageImpl<>(formatResults, pageable, rawResults.getTotalElements());
+
+  }
```

Comment:
Suggestion: Instead of still using the repository, it would be nice to have this do the lookup similar to how the  `PageDescriptionFinder` does it. A row mapper could then be created that would handle mapping from the database entity to a "UI friendly" model.